core/arch/x86/decode_table.c

```diff
@@ -4001,14 +4023,13 @@ const instr_info_t prefix_extensions[][12] = {
     {OP_cvttpd2dq,0x660fe610, "cvttpd2dq", Vdq, xx, Wpd, xx, xx, mrm, x, END_LIST},
     {OP_cvtpd2dq, 0xf20fe610, "cvtpd2dq",  Vdq, xx, Wpd, xx, xx, mrm, x, END_LIST},
     {INVALID,        0x0fe610, "(bad)", xx, xx, xx, xx, xx, no, x, END_LIST},
-    {OP_vcvtdq2pd, 0xf30fe610, "vcvtdq2pd",  Vvd, xx, Wvq_dq, xx, xx, mrm|vex, x, END_LIST},
-    {OP_vcvttpd2dq,0x660fe610, "vcvttpd2dq", Vx, xx, Wvd, xx, xx, mrm|vex, x, END_LIST},
-    {OP_vcvtpd2dq, 0xf20fe610, "vcvtpd2dq",  Vx, xx, Wvd, xx, xx, mrm|vex, x, END_LIST},
-    /* TODO i#1312: Support AVX-512. */
+    {OP_vcvtdq2pd, 0xf30fe610, "vcvtdq2pd",  Vvd, xx, Wvq_dq, xx, xx, mrm|vex, x, tevexw[57][0]},
+    {OP_vcvttpd2dq,0x660fe610, "vcvttpd2dq", Vx, xx, Wvd, xx, xx, mrm|vex, x, tpe[77][10]},
+    {OP_vcvtpd2dq, 0xf20fe610, "vcvtpd2dq",  Vx, xx, Wvd, xx, xx, mrm|vex, x, tpe[77][11]},
     {INVALID,   0x0fe610, "(bad)", xx, xx, xx, xx, xx, no, x, NA},
-    {INVALID, 0xf30fe610, "(bad)", xx, xx, xx, xx, xx, no, x, NA},
-    {INVALID, 0x660fe610, "(bad)", xx, xx, xx, xx, xx, no, x, NA},
-    {INVALID, 0xf20fe610, "(bad)", xx, xx, xx, xx, xx, no, x, NA},
+    {EVEX_W_EXT, 0xf30fe610, "(evex_W ext 57)", xx, xx, xx, xx, xx, mrm|evex, x, 57},
+    {OP_vcvttpd2dq,0x660fe610, "vcvttpd2dq", Ve, xx, KEb, Wed, xx, mrm|evex, x, END_LIST},
+    {OP_vcvtpd2dq, 0xf20fe610, "vcvtpd2dq",  Ve, xx, KEb, Wed, xx, mrm|evex, x, END_LIST},
```

Comment:
Either there's something wrong, or my decoders are not up to date or sthg (a couple years old but they've had all the other AVX-512 opcodes and since they all think this is bad I think there's something wrong):

```
  0x0000557b208acc81  62 81 7f 4f e6 c7    vcvtpd2dq zmm16{, k7}, zmm31
```

```
$ disasm 62 81 7f 4f e6 c7
Disassembling 0x62 0x81 0x7f 0x4f 0xe6 0xc7
<stdin>:1:1: warning: invalid instruction encoding
0x62 0x81 0x7f 0x4f 0xe6 0xc7
^
<stdin>:1:26: warning: invalid instruction encoding
0x62 0x81 0x7f 0x4f 0xe6 0xc7
                         ^
llvm-mc:   0x62 0x81 0x7f 0x4f 0xe6 0xc7 
capstone: 
bfd:       62 81 7f 4f e6 (bad) {%k7}
```

There are a *lot* of instances of this so I suspect others here have problems.

Reply:
xref #3639

Reply:
Problem was not #3639, it's actually "just" the W bit. Fixed. More detailed testing required, I'll set something up.

Reply:
W1 bit was fixed in this patch. #3639 is still unfixed. It mainly affects the register mnemonic that is printed, the instruction create macros and the source's operand size. Please note the bug already existed before AVX-512 in some of the VEX instructions. I will fix but I consider this parallel to this patch, not blocking.